### PR TITLE
Fix duplicate setting

### PR DESCRIPTION
**Pull request by spacecowboy**
Seems like `rotation_delay` has never been an accessible configuration option. It doesn't seem to be in the manual either. Trying to configure the `threshold` option with a valid suffixed value (like `20m`) doesn't work either since `rotation_delay` will try the same value as an integer.

Also added a test to check for duplicates.

### CODE REVIEW

File: community/kernel/src/main/java/org/neo4j/graphdb/factory/GraphDatabaseSettings.java [hunk 1]
```diff
@@ -182,7 +182,8 @@
             setting( "store.interval.log.rotation", DURATION, "10m" );
 
     @Description( "Minimum time (in seconds) after last rotation of the internal log before it may be rotated again." )
-    public static final Setting<Integer> store_internal_log_rotation_delay = setting("store.internal_log.rotation_threshold", INTEGER, "300", min(0), max( Integer.MAX_VALUE ) );
+    public static final Setting<Integer> store_internal_log_rotation_delay = setting("store.internal_log" +
+            ".rotation_delay", INTEGER, "300", min(0), max( Integer.MAX_VALUE ) );
```
Comment (thobe) [inline community/kernel/src/main/java/org/neo4j/graphdb/factory/GraphDatabaseSettings.java]: Could we avoid line-breaking these strings?

**Comment by cleishm** [comment]
While you're at it, fixing the documentation to consistently indicate that the threshold defaults to 250Mb would be nice (http://neo4j.com/docs/stable/configuration-logical-logs.html shows 25Mb).

**Comment by MishaDemianenko** [comment]
@spacecowboy can you please fix @cleishm comment

**Comment by spacecowboy** [comment]
@MishaDemianenko @cleishm done

File: community/kernel/src/main/java/org/neo4j/graphdb/factory/GraphDatabaseSettings.java [hunk 2]
```diff
@@ -182,7 +182,8 @@
             setting( "store.interval.log.rotation", DURATION, "10m" );
 
     @Description( "Minimum time (in seconds) after last rotation of the internal log before it may be rotated again." )
-    public static final Setting<Integer> store_internal_log_rotation_delay = setting("store.internal_log.rotation_threshold", INTEGER, "300", min(0), max( Integer.MAX_VALUE ) );
+    public static final Setting<Integer> store_internal_log_rotation_delay =
+            setting("store.internal_log.rotation_delay", INTEGER, "300", min(0), max( Integer.MAX_VALUE ) );
```
Comment (tinwelint) [inline community/kernel/src/main/java/org/neo4j/graphdb/factory/GraphDatabaseSettings.java]: I think this should be `DURATION` instead if `INTEGER`, so that we can specify f.ex. "10s"

Comment (MishaDemianenko) [inline community/kernel/src/main/java/org/neo4j/graphdb/factory/GraphDatabaseSettings.java]: @tinwelint  for some reason in description it says that it only possible to have seconds here, should that be updated to anything then?

Comment (tinwelint) [inline community/kernel/src/main/java/org/neo4j/graphdb/factory/GraphDatabaseSettings.java]: I see no reason why this wouldn't be a proper duration. It's also backwards compatible since the default unit is seconds. So please update the description as well.

Comment (MishaDemianenko) [inline community/kernel/src/main/java/org/neo4j/graphdb/factory/GraphDatabaseSettings.java]: @tinwelint agree,
@spacecowboy that will also require proper migration of current value from number to duration.

**Comment by tinwelint** [comment]
@spacecowboy @MishaDemianenko no migration needed

File: community/kernel/src/main/java/org/neo4j/graphdb/factory/GraphDatabaseSettings.java (hunk 2 above)
Comment (tinwelint) [inline community/kernel/src/main/java/org/neo4j/graphdb/factory/GraphDatabaseSettings.java]: @spacecowboy @MishaDemianenko no migration needed. DURATION is a superset of INTEGER, right?

Comment (MishaDemianenko) [inline community/kernel/src/main/java/org/neo4j/graphdb/factory/GraphDatabaseSettings.java]: ah, now i see @tinwelint  thx :)

**Comment by spacecowboy** [comment]
@tinwelint @MishaDemianenko Agree with your comments. Changed to duration.

**Comment by MishaDemianenko** [comment]
@spacecowboy compilation failure

**Comment by spacecowboy** [comment]
@MishaDemianenko green now

File: community/kernel/src/main/java/org/neo4j/graphdb/factory/GraphDatabaseSettings.java [hunk 3]
```diff
@@ -181,8 +181,9 @@
     public static final Setting<Long> store_interval_log_rotation_wait_time =
             setting( "store.interval.log.rotation", DURATION, "10m" );
 
-    @Description( "Minimum time (in seconds) after last rotation of the internal log before it may be rotated again." )
-    public static final Setting<Integer> store_internal_log_rotation_delay = setting("store.internal_log.rotation_threshold", INTEGER, "300", min(0), max( Integer.MAX_VALUE ) );
+    @Description( "Minimum time interval after last rotation of the internal log before it may be rotated again." )
```
Comment (MishaDemianenko) [inline community/kernel/src/main/java/org/neo4j/graphdb/factory/GraphDatabaseSettings.java:184]: @spacecowboy  This should produce time in milliseconds, `RotatingFileOutputStreamSupplier` still assume they are seconds  and will try to convert them back to milliseconds by calling `SECONDS.toMillis( rotationDelaySeconds )`

Comment (spacecowboy) [inline community/kernel/src/main/java/org/neo4j/graphdb/factory/GraphDatabaseSettings.java:184]: @MishaDemianenko good catch. Changed it to not convert it. Also the variables were called `second` but actually stored `milliseconds`.

Comment (tinwelint) [inline community/kernel/src/main/java/org/neo4j/graphdb/factory/GraphDatabaseSettings.java:184]: @spacecowboy `RotatingFileOutputStreamSupplier` does the right thing, it's just the `rotationDelaySeconds` field that is badly named.

**Comment by tinwelint** [comment]
To end this ms vs s madness once and for all we should introduce a `Duration` class that is a `long` AND a `TimeUnit` and pass such around instead of a number for time.

File: community/kernel/src/main/java/org/neo4j/graphdb/factory/GraphDatabaseSettings.java (hunk 3 above)
Comment (tinwelint) [inline community/kernel/src/main/java/org/neo4j/graphdb/factory/GraphDatabaseSettings.java:184]: Oh, I was wrong about this.... DURATION returns `ms` data after all.

**Comment by spacecowboy** [comment]
@MishaDemianenko as discussed with @tinwelint , the units are actually correct. IMO we merge this and leave `TimeUnit` for another day. WDYT?

File: community/logging/src/main/java/org/neo4j/logging/RotatingFileOutputStreamSupplier.java [hunk 4]
```diff
@@ -94,38 +92,43 @@ public long getAsLong()
...
      * @param outputFile             The file that the latest {@link OutputStream} should output to
      * @param rotationThresholdBytes The size above which the file should be rotated
-     * @param rotationDelaySeconds   The minimum time (in seconds) after last rotation before the file may be rotated again
+     * @param rotationDelay          The minimum time (ms) after last rotation before the file may be rotated again
      * @param maxArchives            The maximum number of archived output files to keep
      * @param rotationExecutor       An {@link Executor} for performing the rotation
      * @throws IOException If the output file cannot be created
      */
-    public RotatingFileOutputStreamSupplier( FileSystemAbstraction fileSystem, File outputFile, long rotationThresholdBytes, int rotationDelaySeconds, int maxArchives, Executor rotationExecutor ) throws IOException
+    public RotatingFileOutputStreamSupplier( FileSystemAbstraction fileSystem, File outputFile, long rotationThresholdBytes, int rotationDelay, int maxArchives, Executor rotationExecutor ) throws IOException
```
Comment (MishaDemianenko) [inline community/logging/src/main/java/org/neo4j/logging/RotatingFileOutputStreamSupplier.java]: `rotationDelay` should be long instead of int

**Comment by MishaDemianenko** [comment]
@spacecowboy one small comment to be fixed
